app/src/ui/history/compare.tsx

```diff
@@ -171,10 +172,31 @@ export class CompareSidebar extends React.Component<
     const selectedCommit = this.state.selectedCommit
     const commitSHAs = compareState.commitSHAs
 
-    const emptyListMessage =
-      compareState.formState.kind === ComparisonView.None
-        ? 'No history'
-        : 'No commits'
+    let emptyListMessage
+    if (compareState.formState.kind === ComparisonView.Ahead) {
+      const currentlyComparedBranchName =
+        compareState.formState.comparisonBranch.name
+
+      emptyListMessage = (
+        <p>
+          The compared branch (<Ref>{currentlyComparedBranchName}</Ref>) is up
+          to date with your branch
+        </p>
+      )
+    } else if (compareState.formState.kind === ComparisonView.Behind) {
+      const currentlyComparedBranchName =
+        compareState.formState.comparisonBranch.name
+
+      emptyListMessage = (
+        <p>
+          Your branch is up to date with the compared branch (<Ref>
+            {currentlyComparedBranchName}
+          </Ref>)
+        </p>
+      )
+    } else {
+      emptyListMessage = 'No history'
```

Comment:
the only kinds of `formState.kind`s are Ahead and Behind so this can get simplified because I don't think it's possible to hit the "else" case.
if my hunch is right, then the `<p>` tags so that can get pulled up to the component that puts in `{emptyListMessage}` and leaves us with a simpler ternary

Reply:
There are actually 3 options for `Form.kind`: `none`, `ahead`, `behind`.  You could use a nested ternary instead, but I have opinions about those.